src/api.js

```diff
@@ -13,18 +12,30 @@ export const getArticles = () => {
     })
     .catch((error) => {
       console.log(error, "< error in api.js catch");
+      throw error;
     });
 };
 
 export const getArticleById = (article_id) => {
-  console.log("article fetch by id invoked");
   return newsBaseUrl
     .get(`/articles/${article_id}`)
     .then(({ data }) => {
-      console.log(data, "<single article data return in util");
       return data;
     })
     .catch((error) => {
       console.log(error, "< error in single article fetch");
+      throw error;
+    });
+};
+
+export const getArticleComments = (article_id) => {
+  return newsBaseUrl
+    .get(`/articles/${article_id}/comments`)
+    .then(({ data }) => {
+      return data;
+    })
+    .catch((error) => {
```

Comment:
Unclear if you've had this as feedback already. Its not ideal to have the catch in the api call, as if something does get caught you won't be able to access it in the useEffect as it has been resolved in this file. You can return the data and move the catch to where ever is using this function so you can appropriately handle errors later on.